code/controllers/subsystem/research.dm

```diff
@@ -0,0 +1,72 @@
+
+SUBSYSTEM_DEF(research)
+	name = "Research"
+	flags = SS_KEEP_TIMING
+	priority = 15		//My powergame is priority.
+	wait = 10
+	init_order = INIT_ORDER_RESEARCH
+	var/list/invalid_design_ids = list()		//associative id = number of times
+	var/list/invalid_node_ids = list()			//associative id = number of times
+	var/list/invalid_node_boost = list()		//associative id = error message
+	var/list/obj/machinery/rnd/server/servers = list()
+	var/datum/techweb/science/science_tech
+	var/datum/techweb/admin/admin_tech
+	var/list/techweb_nodes = list()				//associative id = node datum
+	var/list/techweb_categories = list()		//category name = list(node.id = node)
+	var/list/techweb_designs = list()			//associative id = node datum
+	var/list/techweb_nodes_starting = list()	//associative id = node datum
+	var/list/techweb_boost_items = list()		//associative double-layer path = list(node = point_discount)
+	var/list/techweb_nodes_hidden = list()		//Nodes that should be hidden by default.
+	var/list/techweb_point_items = list()		//path = value
+	//----------------------------------------------
+	var/single_server_income = 40.7
+	var/multiserver_calculation = FALSE
+	var/last_income = 0
+	//^^^^^^^^ ALL OF THESE ARE PER SECOND! ^^^^^^^^
+
+	//Aiming for 1.5 hours to max R&D
+	//[88nodes * 5000points/node] / [1.5hr * 90min/hr * 60s/min]
+	//Around 450000 points max???
+
+	var/bomb_research_point_scaling = 1800
+
+/datum/controller/subsystem/research/Initialize()
+	initialize_all_techweb_designs()
+	initialize_all_techweb_nodes()
+	science_tech = new /datum/techweb/science
+	admin_tech = new /datum/techweb/admin
+	autosort_categories()
+	return ..()
+
+/datum/controller/subsystem/research/fire()
+	handle_research_income()
+
+/datum/controller/subsystem/research/proc/handle_research_income()
+	var/bitcoins = 0
+	if(multiserver_calculation)
+		var/eff = calculate_server_coefficient()
+		for(var/obj/machinery/rnd/server/miner in servers)
+			bitcoins += (miner.mine() * eff)	//SLAVE AWAY, SLAVE.

```

Comment:
`(miner.mine() * eff) + (miner2.mine() * eff)` is the same as `(miner.mine() + miner2.mine()) * eff`

Also if you ever have a miner that gets more bitcoins than another this won't scale right.